keras/layers/convolutional.py

```diff
@@ -129,7 +129,13 @@ def build(self, input_shape):
             raise ValueError('The channel dimension of the inputs '
                              'should be defined. Found `None`.')
         input_dim = input_shape[channel_axis]
-        kernel_shape = self.kernel_size + (input_dim, self.filters)
+
+        if self.data_format == 'channels_first':
+            kernel_shape = (self.filters, input_dim) + self.kernel_size
+        else:
+            kernel_shape = self.kernel_size + (input_dim, self.filters)
```

Comment:
Reverted.

But, the major concern here would be a massive performance hit due to extra transpose operation on kernels for backends with "channels_last" format.

Is there any way to avoid it? Say for example, recommending channels_last format for MXNet backend in keras.json config?

Reply:
The same concern exists for Theano with `data_format='channels_last'`.

I do not know the extent of the performance hit. My guess is that it is small. I note that [native TF ops](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d) use the same kernel shape for both data formats, which indicates that performance optimization can probably be handled at the backend level.